frontend/js/src/components/SyndicationFeedModal.tsx

```diff
@@ -0,0 +1,161 @@
+import NiceModal, { useModal } from "@ebay/nice-modal-react";
+import * as React from "react";
+import { useState } from "react";
+
+export type SyndicationFeedModalProps = {
+  feedTitle: string;
+  options: {
+    label: string;
+    key: string;
+    type: "dropdown" | "number" | string;
```

Comment:
I learned recently that React has a type for this specific use case:
```suggestion
    type: React.HTMLInputTypeAttribute;
```

In this case, it won't prevent the type to be an arbitrary string with a typo, but at least autocompletion will provide common use-cases.


A quick note with more advanced typescript, there is a way to conditionally define some props based on another prop, for example requiring `min` and `max` only if `type` = `number`
I don't think it is required to do that here but thought that might be of interest to you:
https://github.com/metabrainz/listenbrainz-server/blob/f346f6aa994acfbcc616c233f8a581ef303ae6a8/frontend/js/src/utils/SearchTrackOrMBID.tsx#L22-L40